pkg/resource/deploy/step.go

```diff
@@ -945,7 +982,20 @@ func (s *ImportStep) Diffs() []resource.PropertyKey                { return s.di
 func (s *ImportStep) DetailedDiff() map[string]plugin.PropertyDiff { return s.detailedDiff }
 
 func (s *ImportStep) Apply(preview bool) (resource.Status, StepCompleteFunc, error) {
-	complete := func() { s.reg.Done(&RegisterResult{State: s.new}) }
+	complete := func() {
+		// Import takes a resource that Pulumi did not create and imports it into pulumi state.
+		now := time.Now().UTC()
+		s.new.Modified = &now
+		// The Created timestamp is unknown as the resource was not provisioned by pulumi.
+		s.new.Created = nil
+
+		if s.old != nil && s.new.ID == s.old.ID {
+			// If there exists an old timestamp, and the IDs are the same, preserve the timestamps.
+			s.new.Created = s.old.Created
+			s.new.Modified = s.old.Modified
```

Comment:
I think this will just be setting these to nil each time because we set s.old below (line 1043) and don't give a value for Created or Modified.

Reply:
Will remove the check then.